### PR TITLE
Improve S4653 ('unit-no-unknown'): Support ignoring any kind of unit

### PR DESCRIPTION
**Pull request by yassin-kammoun-sonarsource**
Fixes https://github.com/SonarSource/sonar-css/pull/367

### CODE REVIEW

**Review by vilchik-elena — CHANGES_REQUESTED** [review]
I didn't find any place describing why we need this parameter. Also why default should be `x`

File: sonar-javascript-plugin/src/main/java/org/sonar/plugins/javascript/css/rules/UnitNoUnknown.java [hunk 1]
```diff
@@ -22,22 +22,37 @@
...
+import static org.sonar.plugins.javascript.css.rules.RuleUtils.splitAndTrim;
 
 @Rule(key = "S4653")
 public class UnitNoUnknown implements CssRule {
 
+  private static final String DEFAULT_IGNORED_UNITS = "x";
+
+  @RuleProperty(
+    key = "ignoredUnits",
+    description = "Comma-separated list of \"units\" to consider as valid.",
```
Comment (vilchik-elena) [inline sonar-javascript-plugin/src/main/java/org/sonar/plugins/javascript/css/rules/UnitNoUnknown.java:36]: should we mention that regex are supported?

File: sonar-javascript-plugin/src/main/java/org/sonar/plugins/javascript/css/rules/UnitNoUnknown.java [hunk 2]
```diff
@@ -22,22 +22,37 @@
...
   }
 
   @Override
   public List<Object> stylelintOptions() {
-    return Arrays.asList(true, new StylelintIgnoreOption());
+    return Arrays.asList(true, new StylelintIgnoreOption(splitAndTrim(ignoredUnits)));
   }
 
   private static class StylelintIgnoreOption {
     // Used by GSON serialization
```
[the remark this answers is not in the document]
Comment (vilchik-elena) [reply sonar-javascript-plugin/src/main/java/org/sonar/plugins/javascript/css/rules/UnitNoUnknown.java:51]: Ok, I see now that we use GSON when sending the request

File: sonar-javascript-plugin/src/main/java/org/sonar/plugins/javascript/css/rules/UnitNoUnknown.java (hunk 1 above)
Comment (yassin-kammoun-sonarsource) [reply sonar-javascript-plugin/src/main/java/org/sonar/plugins/javascript/css/rules/UnitNoUnknown.java:36]: We should, similarly to what you suggested in the other PR.

**Comment by vilchik-elena** [comment]
> Which parameter are you referring to?

The parameter you have added in this PR :)

**Comment by yassin-kammoun-sonarsource** [comment]
Closing this PR as discussed.

**Pull request closed without merging**